### PR TITLE
Add architecture to package name on windows

### PR DESCRIPTION
**Pull request by jaym**
Potentially breaking change?

cc @chef/omnibus-maintainers @ksubrama @chefsalim @thommay

### CODE REVIEW

**Comment by jaym** [comment]
dont' know if we want this in 4.1

File: lib/omnibus/packagers/msi.rb [hunk 1]
```diff
@@ -321,11 +321,11 @@ def package_name
     end
 
     def msi_name
-      "#{project.package_name}-#{project.build_version}-#{project.build_iteration}.msi"
+      "#{project.package_name}-#{project.build_version}-#{project.build_iteration}.#{Config.windows_arch}.msi"
```
Comment (schisamo) [inline lib/omnibus/packagers/msi.rb]: Looking at how other Windows packages are named ([Ruby](http://rubyinstaller.org/downloads/), [Git](https://git-scm.com/download/win) and [various Chocolatey stuff](https://chocolatey.org/packages?q=64-bit)) I think we should separate iteration and arch with a dash (`-`).

**Comment by schisamo** [comment]
I'm not sure I would call this a breaking change but my arm could be twisted if a good case is made.

File: lib/omnibus/packagers/msi.rb (hunk 1 above)
Comment (sethvargo) [inline lib/omnibus/packagers/msi.rb]: :+1:

**Comment by sethvargo** [comment]
I think this is fine to put into 4.1 as long as it's called out in the changelog. It isn't breaking any APIs, just the output format. As long as it's clearly called out in the CL, I think it's fine.

**Comment by ksubrama** [comment]
:+1:

**Comment by yzl** [comment]
:+1:

**Comment by jaym** [comment]
cc @chef/omnibus-maintainers

**Comment by schisamo** [comment]
:+1:

**Pull request merged**